File: DemoApp/DemoApp/CollectionView/BaseCollectionViewController.m
//
//  BaseCollectionViewController.m
//  DemoApp
//
//  Created by Glispa GmbH on 11.08.16.
//  Copyright © 2016 Glispa GmbH All rights reserved.
//

#import "BaseCollectionViewController.h"
#import "AppControllerManagement.h"

@implementation BaseCollectionViewController

@end


@interface BaseCollectionViewController (AppManagement) <AppControllerManagement>
@end

@implementation BaseCollectionViewController (AppManagement)

- (void)applyUserData:(NSDictionary <NSString *, NSObject *> *)userData {
    if ([userData[@"adUnitId"] isKindOfClass:[NSString class]]) {
        self.adUnitId = (NSString *)userData[@"adUnitId"];
    }
}

@end

File: DemoApp/DemoAppSwift/BaseCollectionViewController.swift
//
//  BaseCollectionViewController.swift
//  DemoApp
//
//  Created by Glispa GmbH on 12.08.16.
//  Copyright © 2016 Glispa GmbH All rights reserved.
//

import UIKit
import AvocarrotNativeView

class BaseCollectionViewController: UIViewController {
    @IBOutlet weak var collectionView: UICollectionView!

    public var adUnitId: String!
}

extension BaseCollectionViewController: AppControllerManagement {
    func applyUserData(_ userData: [String:Any]) {
        if let adId = userData["adUnitId"] as? String {
            adUnitId = adId
        }
    }
}
